common/src/main/java/com/wynntils/core/text/PartStyle.java

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.text;
+
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.mc.ComponentUtils;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.Style;
+
+public final class PartStyle {
+    private static final String STYLE_PREFIX = "§";
+
+    private final StyledTextPart owner;
+
+    private final CustomColor color;
+    private final boolean obfuscated;
+    private final boolean bold;
+    private final boolean strikethrough;
+    private final boolean underlined;
+    private final boolean italic;
+    private final ClickEvent clickEvent;
+    private final HoverEvent hoverEvent;
+
+    private PartStyle(
+            StyledTextPart owner,
+            CustomColor color,
+            boolean obfuscated,
+            boolean bold,
+            boolean strikethrough,
+            boolean underlined,
+            boolean italic,
+            ClickEvent clickEvent,
+            HoverEvent hoverEvent) {
+        this.owner = owner;
+        this.color = color;
+        this.obfuscated = obfuscated;
+        this.bold = bold;
+        this.strikethrough = strikethrough;
+        this.underlined = underlined;
+        this.italic = italic;
+        this.clickEvent = clickEvent;
+        this.hoverEvent = hoverEvent;
+    }
+
+    PartStyle(PartStyle partStyle, StyledTextPart owner) {
```

Comment:
Just checking, this is not private since it is used from another class?

Reply:
Yeah, everything that is package private is used somewhere